### PR TITLE
refactor(knobs): Gallery optional/mandatory knobs -FRONT-926

### PR DESCRIPTION
**Pull request by Joosthe**
# PR description

Please drop a few lines about the PR: what it does, how to test it, etc.

## QA Checklist

In order to ensure a safe and quick review, please check that your PR follow those guidelines:

- [ ] I have put the vanilla component as `devDependencies`
- [ ] I have put the specs package as `devDependencies`
- [ ] I have added the components directly used in the twig file (with `include` or `embed`) as `dependencies`
- [ ] My component is listed in `@ecl-twig/ec-components`'s `dependencies`
- [ ] My variables naming follow the guidelines (snake case for twig)
- [ ] I have provided tests
- [ ] I have provided documentation (for the "notes" tab)
- [ ] If my local `yarn.lock` contains changes, I have committed it
- [ ] I have given my PR the proper label (`pr: review needed` to indicate that I'm done and now waiting for a review ,`pr: wip` to indicate that I'm actively working on it ...)

### CODE REVIEW

File: src/ec/packages/ec-component-gallery/gallery.story.js [hunk 1]
```diff
@@ -1,29 +1,77 @@
...
-  }
-});
-data.overlay.close.icon.path = defaultSprite;
-data.overlay.previous.icon.path = defaultSprite;
-data.overlay.next.icon.path = defaultSprite;
-data.overlay.download.icon.path = defaultSprite;
-data.overlay.share.icon.path = defaultSprite;
+const preperaGallery = data => {
+  data.items.forEach((item, index) => {
+    item.share_path = select(
```
Comment (papegaill) [inline src/ec/packages/ec-component-gallery/gallery.story.js]: What about the other parameters?
- "path"
- "alt"
- "meta"
 - "icon"

File: src/ec/packages/ec-component-gallery/gallery.story.js [hunk 2]
```diff
@@ -1,29 +1,77 @@
...
+    defaultSprite,
+    buttonLabels.required
+  );
+  data.overlay.next.icon.path = select(
+    'overlay.next.icon.path',
+    [defaultSprite],
+    defaultSprite,
+    buttonLabels.required
+  );
+  data.overlay.download.icon.path = select(
```
Comment (papegaill) [inline src/ec/packages/ec-component-gallery/gallery.story.js]: this is a link object, not only the icon.path should be a knob

File: src/ec/packages/ec-component-gallery/gallery.story.js [hunk 3]
```diff
@@ -1,29 +1,77 @@
...
+    'overlay.download.icon.path',
+    [defaultSprite],
+    defaultSprite,
+    buttonLabels.required
+  );
+  data.overlay.share.icon.path = select(
```
Comment (papegaill) [inline src/ec/packages/ec-component-gallery/gallery.story.js]: this is a link object, not only the icon.path should be a knob

File: src/ec/packages/ec-component-gallery/gallery.story.js (hunk 1 above)
Comment (Joosthe) [reply src/ec/packages/ec-component-gallery/gallery.story.js]: i didn't see them in the documentation in ecl for the gallery but i will put them in

**Pull request merged**